### PR TITLE
scheduler: three per-pass CPU micro-cuts from a py-spy wall profile

### PR DESCRIPTION
**Pull request by sshleifer**
Three small scheduler-thread cuts found in a py-spy wall profile at ~1,500-request batches (the scheduler thread was ~99% busy):

- `_stream_output_generation`: fuse the four `any()` flag scans over the batch into one pass (~2.1% of thread time).
- `_maybe_collect_customized_info`: gate at the call sites on `logits_output.customized_info is not None` — the per-request call overhead alone is measurable (~1.4%).
- `record_batch_in_overlap`: cache `ScheduleBatch` field names once instead of `dataclasses.fields()` reflection per pass (~1.0%).

No behavior change. Part of a scheduler-CPU series measured on an internal multi-turn RL rolling benchmark (1x GB300).

---
### CI States

Latest PR Test (Base): :x: [Run #32896525222](https://github.com/sgl-project/sglang/actions/runs/32896525222)
Latest PR Test (Extra): :x: [Run #32896524777](https://github.com/sgl-project/sglang/actions/runs/32896524777)
Latest PR Test (AMD ROCm 7.2): :x: [Run #32896525060](https://github.com/sgl-project/sglang/actions/runs/32896525060)

### CODE REVIEW

File: python/sglang/srt/managers/scheduler.py [hunk 1]
```diff
@@ -3660,16 +3660,20 @@ def update_running_batch(self, batch: ScheduleBatch) -> Optional[ScheduleBatch]:
         batch.prepare_for_decode()
         return batch
 
+    _schedule_batch_field_names: tuple[str, ...] | None = None
```
Comment (sshleifer) [inline python/sglang/srt/managers/scheduler.py:3663]: ScheduleBatch field names, resolved once per process: `dataclasses.fields()` reflection is on the per-pass critical path.

File: python/sglang/srt/managers/scheduler_components/batch_result_processor.py [hunk 2]
```diff
@@ -1165,7 +1169,8 @@ def _handle_finish_state_updated_req(
 
             req.time_stats.set_completion_time()
 
-        self._maybe_collect_customized_info(i, req, logits_output)
+        if logits_output is not None and logits_output.customized_info is not None:
```
Comment (sshleifer) [inline python/sglang/srt/managers/scheduler_components/batch_result_processor.py:1172]: `_maybe_collect_customized_info` is gated at the call site (here and in the prefill path above): the per-request call overhead alone is measurable at large batches.

File: python/sglang/srt/managers/scheduler_components/output_streamer.py [hunk 3]
```diff
@@ -144,18 +144,21 @@ def _stream_output_generation(
...
-        return_routed_experts = any(
-            req.return_routed_experts for req in reqs if req is not skip_req
-        )
-        return_indexer_topk = any(
-            req.return_indexer_topk for req in reqs if req is not skip_req
-        )
-        return_sampling_mask = any(
-            req.return_sampling_mask for req in reqs if req is not skip_req
-        )
+        return_hidden_states = False
```
Comment (sshleifer) [inline python/sglang/srt/managers/scheduler_components/output_streamer.py:147]: One fused scan instead of four `any()` passes over the running batch.

**Review by sshleifer — COMMENTED** [review]
Moved the PR's inline rationale comments into the review annotations below, anchored to the lines they described, so the diff itself stays comment-free.

-Robot

**Comment by sshleifer** [comment]
Withdrawing: same reasoning as #36365 — redundant on our fork once the scheduler overlaps prefill replay, and dropped internally. Still valid without that overlap; happy to reopen if there's interest.
